lib/active_model/serializer/caching.rb

```diff
@@ -0,0 +1,106 @@
+      #  AS
+      #  c/git/emberjs/ember-crm-backend/app/serializers/lead_serializer.rb
+      CALLER_FILE = /
+        \A       # start of string
+        .+       # file path (one or more characters)
+        (?=      # stop previous match when
+          :\d+     # a colon is followed by one or more digits
+          :in      # followed by a colon followed by in
+        )
+      /x
+
+      module ClassMethods
+        def inherited(base)
+          super
+          caller_line = caller.second
+          base._cache_digest = digest_caller_file(caller_line)
+        end
+
+        # Hashes contents of file for +_cache_digest+
+        def digest_caller_file(caller_line)
+          serializer_file_path = caller_line[CALLER_FILE]
+          serializer_file_contents = IO.read(serializer_file_path)
+          Digest::MD5.hexdigest(serializer_file_contents)
+        rescue TypeError, Errno::ENOENT
+          warn <<-EOF.strip_heredoc
+            Cannot digest non-existent file: '#{caller_line}'.
+            Please set `::_cache_digest` of the serializer
+            if you'd like to cache it.
+            EOF
+          ''.freeze
+        end
+
+        # @api private
+        # Used by FragmentCache on the CachedSerializer
+        #  to call attribute methods on the fragmented cached serializer.
+        def fragmented(serializer)
+          self._fragmented = serializer
+        end
+
+        # Enables a serializer to be automatically cached
+        #
+        # Sets +::_cache+ object to <tt>ActionController::Base.cache_store</tt>
+        #   when Rails.configuration.action_controller.perform_caching
+        #
+        # @params options [Hash] with valid keys:
+        #   key            : @see ::_cache_key
+        #   only           : @see ::_cache_only
+        #   except         : @see ::_cache_except
+        #   skip_digest    : does not include digest in cache_key
+        #   all else       : @see ::_cache_options
+        #
+        # @example
+        #   class PostSerializer < ActiveModel::Serializer
+        #     cache key: 'post', expires_in: 3.hours
+        #     attributes :title, :body
+        #
+        #     has_many :comments
+        #   end
+        #
+        # @todo require less code comments. See
+        # https://github.com/rails-api/active_model_serializers/pull/1249#issuecomment-146567837
+        def cache(options = {})
+          self._cache = ActiveModelSerializers.config.cache_store if ActiveModelSerializers.config.perform_caching
+          self._cache_key = options.delete(:key)
+          self._cache_only = options.delete(:only)
+          self._cache_except = options.delete(:except)
+          self._cache_options = (options.empty?) ? nil : options
+        end
+
+        # Used to cache serializer name => serializer class
+        # when looked up by Serializer.get_serializer_for.
+        def serializers_cache
+          @serializers_cache ||= ThreadSafe::Cache.new
+        end
```

Comment:
`serializers_cache` should probably be in the same file as `get_serializer_for`, right?

```ruby
     # @api private
     # Find a serializer from a class and caches the lookup.
     # Preferentially retuns:
     #   1. class name appended with "Serializer"
     #   2. try again with superclass, if present
     #   3. nil
     def self.get_serializer_for(klass)
       return nil unless config.serializer_lookup_enabled
       serializers_cache.fetch_or_store(klass) do
```